apps/client/src/components/globalSettings/UnivariateCellPlot.vue

```diff
@@ -0,0 +1,288 @@
+            ></div>
+            <div
+                class="selection-box left"
+                style="position: absolute; bottom: 12px"
+            >
+                <input
+                    type="text"
+                    :value="minValue"
+                    @input="updateMinValue($event.target.value)"
+                    @keyup.enter="handleEnter($event)"
+                    @blur="applyManualFilter()"
+                    @focus="$event.target.select()"
+                    class="minMax"
+                />
+            </div>
+            <div
+                class="selection-box right"
+                style="position: absolute; bottom: 12px; right: 0"
+            >
+                <input
+                    type="text"
+                    :value="maxValue"
+                    @input="updateMaxValue($event.target.value)"
+                    @keyup.enter="handleEnter($event)"
+                    @blur="applyManualFilter()"
+                    @focus="$event.target.select()"
+                    class="minMax"
+                />
+            </div>
+        </q-item-section>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { ref, watch } from 'vue';
+import type { PropType } from 'vue';
+import * as vg from '@uwdata/vgplot';
+import { useCellMetaData } from '@/stores/cellMetaData';
+import { useDatasetSelectionStore } from '@/stores/datasetSelectionStore';
+import { useSelectionStore } from '@/stores/selectionStore';
+import { storeToRefs } from 'pinia';
+
+const vgPlotContainer = ref<HTMLDivElement | null>(null);
+const cellMetaData = useCellMetaData();
+const datasetSelectionStore = useDatasetSelectionStore();
+const { dataInitialized } = storeToRefs(cellMetaData);
+const selectionStore = useSelectionStore();
+const { Selections } = storeToRefs(selectionStore);
+const emit = defineEmits(['selectionChange']);
+const props = defineProps({
+    plotName: {
+        type: String as PropType<string>,
+        required: true,
+    },
+    plotBrush: {
+        type: Object as PropType<vg.Selection>,
+        required: true,
+    },
+});
+
+const minValue = ref('min');
+const maxValue = ref('max');
+
+const updateMinValue = (value: string) => {
+    minValue.value = value;
+};
+
+const updateMaxValue = (value: string) => {
+    maxValue.value = value;
+};
+
+const handleEnter = (event: KeyboardEvent) => {
+    //event.preventDefault(); // Prevent default form submission on Enter
+    applyManualFilter();
+    //event.target.blur(); // Manually blur the input element
+};
+
+// Called by applyManualFilter when min and max textbox values are entered.
+const updateBrushSelection = (min: number, max: number) => {
+    console.log(props.plotBrush);
+    if (props.plotBrush) {
+        const clause = {
+            source: props.plotName,
+            value: [min, max],
+            predicate: `${props.plotName} BETWEEN ${min} AND ${max}`,
+        };
+
+        props.plotBrush.activate(clause);
```

Comment:
You have already listed some issues with the manual filter selection. I haven't looked into this deeply, but I would expect that you have to update the existing clause, not activate a new one?